src/agent/src/windows_service.cpp

```diff
@@ -0,0 +1,315 @@
+#include <windows_service.hpp>
+
+#include <agent.hpp>
+#include <logger.hpp>
+#include <signal_handler.hpp>
+
+#include <memory>
+#include <unordered_map>
+
+SERVICE_STATUS g_ServiceStatus = {};
+SERVICE_STATUS_HANDLE g_StatusHandle = nullptr;
+HANDLE g_ServiceStopEvent = INVALID_HANDLE_VALUE;
```

Comment:
I believe that these could go into the anonymous namespace, as a safety precaution.

Reply:
Done!